src/Orleans.Core/CodeGeneration/GenericMethodInvoker.cs

```diff
@@ -114,6 +111,8 @@ private GenericMethodInvokerDelegate CreateInvoker(object[] arguments)
                 $"GenericMethodInvoker_{this.grainInterfaceType}_{concreteMethod.Name}",
                 GenericMethodInvokerDelegateMethodInfo);
 
+            var temp = il.DeclareLocal(typeof(ValueTask<object>));
```

Comment:
It's been a while since I've worked closely with IL, but should this be of type `returnType` and be moved into the respective branches `if (returnType == typeof(ValueTask))` and its sibling?

Reply:
You're 100% correct. Will fix.

Reply:
Done.